path_role_test.go

```diff
@@ -311,3 +311,70 @@ func TestPath_Delete(t *testing.T) {
 		t.Fatalf("Unexpected resp data: expected nil got %#v\n", resp.Data)
 	}
 }
+
+func TestPath_Migration(t *testing.T) {
+	b, storage := getBackend(t)
+
+	// Define role entry as it would look if stored with version prior to Vault 1.9.0.
+	// Note: it does not have field "alias_name_source".
+	entryStr := `{
+		"token_bound_cidrs": null,
+		"token_explicit_max_ttl": 0,
+		"token_max_ttl": 0,
+		"token_no_default_policy": false,
+		"token_num_uses": 0,
+		"token_period": 0,
+		"token_policies": null,
+		"token_type": 0,
+		"token_ttl": 0,
+		"bound_service_account_names": [
+		  "name"
+		],
+		"bound_service_account_namespaces": [
+		  "namespaces"
+		],
+		"audience": "",
+		"policies": null,
+		"num_uses": 0,
+		"ttl": 0,
+		"max_ttl": 0,
+		"period": 0,
+		"BoundCIDRs": null
+	}`
+
+	entry := logical.StorageEntry{
+		Key:   "role/old-entry",
+		Value: []byte(entryStr),
+	}
+
+	// Store the role entry.
+	err := storage.Put(context.Background(), &entry)
+	if err != nil {
+		t.Fatalf("Could not store role entry: %s\n", err)
+	}
+
+	// Read the role that was stored with older version of Vault.
+	req := &logical.Request{
+		Operation: logical.ReadOperation,
+		Path:      "role/old-entry",
+		Storage:   storage,
+	}
+
+	resp, err := b.HandleRequest(context.Background(), req)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("err:%s resp:%#v\n", err, resp)
+	}
+
+	// Writing the role back should succeed.
```

Comment:
Just to tighten the test a little, is it maybe worth deserialising `resp.Data` into `roleStorageEntry` as well so that we can assert `AliasNameSource` is set to the default we would expect?

Reply:
Good suggestion, now fixed!